llvm/test/CodeGen/X86/yk-insert-stackmaps.ll

```diff
@@ -0,0 +1,59 @@
+; RUN: llc -stop-after=yk-stackmaps --yk-insert-stackmaps < %s  | FileCheck %s
+
+; Check whether there is llvm.experimental.stackmaps section added after
+; normal calls but not after external calls
+
+; CHECK-LABEL: define dso_local i32 @main
+; CHECK: %call = call i32 @foo(i32 noundef %0)
+; CHECK-NEXT: call void (i64, i32, ...) @llvm.experimental.stackmap(i64 0, i32 0, ptr %argc.addr, ptr %a, ptr %b, i32 %0)
+; CHECK: %call1 = call i32 @foo(i32 noundef %1)
+; CHECK-NEXT: call void (i64, i32, ...) @llvm.experimental.stackmap(i64 1, i32 0, ptr %argc.addr, ptr %a, ptr %b, i32 %1)
+; CHECK: %call2 = call i32 (ptr, ...) @printf(ptr noundef @.str, i32 noundef %2, i32 noundef %3, i32 noundef %4)
+; CHECK-NOT: call void (i64, i32, ...) @llvm.experimental.stackmap
+
+@.str = private unnamed_addr constant [10 x i8] c"%d %d %d\0A\00", align 1
+
+; Function Attrs: noinline nounwind uwtable
+define dso_local i32 @foo(i32 noundef %x) #0 {
+entry:
+  %x.addr = alloca i32, align 4
+  store i32 %x, ptr %x.addr, align 4
+  %0 = load i32, ptr %x.addr, align 4
+  %add = add nsw i32 %0, 10
+  ret i32 %add
+}
+
+; Function Attrs: noinline nounwind uwtable
+define dso_local i32 @main(i32 noundef %argc, ptr noundef %argv) #0 {
+entry:
+  %retval = alloca i32, align 4
+  %argc.addr = alloca i32, align 4
+  %argv.addr = alloca ptr, align 8
+  %a = alloca i32, align 4
+  %b = alloca i32, align 4
+  store i32 0, ptr %retval, align 4
+  store i32 %argc, ptr %argc.addr, align 4
+  store ptr %argv, ptr %argv.addr, align 8
+  %0 = load i32, ptr %argc.addr, align 4
+  %call = call i32 @foo(i32 noundef %0)
+  call void (i64, i32, ...) @llvm.experimental.stackmap(i64 0, i32 0, ptr %argc.addr, ptr %a, ptr %b, i32 %0)
```

Comment:
These calls shouldn't be in the IR as they'll be added by the stackmap pass.

Reply:
corrected here: d555672c3e452f38019cb7dbedb3ca46f5aefcda